COLLABORATOR_GUIDE.md

```diff
@@ -176,6 +176,10 @@ All pull requests must pass continuous integration tests. Code changes must pass
 on [project CI server](https://ci.nodejs.org/). Pull requests that only change
 documentation and comments can use Travis CI results.
 
+Travis CI jobs have a fixed running time limit and building Node.js sometimes
+exceeds this. If the `Compile Node.js` Travis CI job fails in around 45 minutes
+with an exit code of 143 in the job log restart the job.
```

Comment:
Clear up a little to let people know why the restart may help.
```suggestion
with an exit code of 143 (presumably due to timeout) in the job log restart the job.
When Travis CI job fails with a timeout it still saves `ccache` data which will
speed up the consequent build and it may finish in time.
```

Reply:
Reworded, PTAL. Ping @Trott (again) since I'm adding more words here and making this more verbose.

Reply:
How about this (or something like it)?

```markdown
Travis CI jobs have a fixed running time limit that building Node.js sometimes
exceeds. If the Compile Node.js Travis CI job has timed out, it will fail after
around 45 minutes. The exit code will be 143, indicating that a SIGTERM signal
terminated the make command. When this happens, restart the timed out job. It
will reuse built artifacts from the previous timed-out run, and thus take less
time to complete.
```